enterprise/coderd/prebuilds/metricscollector_test.go

```diff
@@ -1,73 +1,247 @@
+								}
+							})
+							db, pubsub := dbtestutil.NewDB(t)
+							reconciler := prebuilds.NewStoreReconciler(db, pubsub, codersdk.PrebuildsConfig{}, logger, quartz.NewMock(t))
+							ctx := testutil.Context(t, testutil.WaitLong)
+
+							createdUsers := []uuid.UUID{prebuilds.OwnerID}
+							for _, user := range slices.Concat(test.ownerIDs, test.initiatorIDs) {
+								if !slices.Contains(createdUsers, user) {
+									dbgen.User(t, db, database.User{
+										ID: user,
+									})
+									createdUsers = append(createdUsers, user)
+								}
+							}
+
+							collector := prebuilds.NewMetricsCollector(db, logger, reconciler)
+							registry := prometheus.NewPedanticRegistry()
+							registry.Register(collector)
+
+							numTemplates := 2
+							for i := 0; i < numTemplates; i++ {
+								orgID, templateID := setupTestDBTemplate(t, db, ownerID)
+								templateVersionID := setupTestDBTemplateVersion(t, ctx, db, pubsub, orgID, ownerID, templateID)
+								preset := setupTestDBPreset(t, ctx, db, pubsub, templateVersionID, 1, uuid.New().String())
+								setupTestDBPrebuild(
+									t, ctx, db, pubsub,
+									transition, jobStatus, orgID, templateID, templateVersionID, preset.ID, initiatorID, ownerID,
+								)
+							}
+
+							metricsFamilies, err := registry.Gather()
+							require.NoError(t, err)
+
+							templates, err := db.GetTemplates(ctx)
+							require.NoError(t, err)
+							require.Equal(t, numTemplates, len(templates))
+
+							for _, template := range templates {
+								templateVersions, err := db.GetTemplateVersionsByTemplateID(ctx, database.GetTemplateVersionsByTemplateIDParams{
+									TemplateID: template.ID,
+								})
+								require.NoError(t, err)
+								require.Equal(t, 1, len(templateVersions))
+
+								presets, err := db.GetPresetsByTemplateVersionID(ctx, templateVersions[0].ID)
+								require.NoError(t, err)
+								require.Equal(t, 1, len(presets))
+
+								for _, preset := range presets {
+									checks := []metricCheck{
+										{"coderd_prebuilds_created", test.shouldIncrementPrebuildsCreated, true},
+										{"coderd_prebuilds_failed", test.shouldIncrementPrebuildsFailed, true},
+										{"coderd_prebuilds_assigned", test.shouldIncrementPrebuildsAssigned, true},
+										{"coderd_prebuilds_desired", test.shouldSetDesiredPrebuilds, false},
+										{"coderd_prebuilds_actual", test.shouldSetActualPrebuilds, false},
+										{"coderd_prebuilds_eligible", test.shouldSetEligiblePrebuilds, false},
+									}
+
+									labels := map[string]string{
+										"template_name": template.Name,
+										"preset_name":   preset.Name,
+									}
+
+									for _, check := range checks {
+										metric := findMetric(metricsFamilies, check.name, labels)
+										if check.value == nil {
+											continue
+										}
+
+										require.NotNil(t, metric, "metric %s should exist", check.name)
+										if check.isCounter {
```

Comment:
I _think_ this is where `promtestutil.ToFloat64` can help simplify things.

Reply:
Tried it. Didn't work because the metric here doesn't implement the collector interface. I checked the function documentation and they discourage using it pretty heavily. Going to stick to what I have for now if thats okay?